CI: add check+clippy+audit

rursprung opened the pull request:
see the individual commit messages for further details.

@mgautierfr: i noticed that so far your PR settings do not seem to enforce that the CI actually passes. since you're using a build matrix it's cumbersome to do this manually (you'd have to list all combinations of the matrix and update it every time you change that). i've started using the `build-results` job which i've added here in most of my rust repos nowadays and it works fine. once the PR is merged you can just list this one job as being required for PRs to be merged and then you're set.

mgautierfr commented:
It seems clippy is not installed on nightly in the CI.
But I think it would be better to run clippy only on stable (or make it not mandatory). Nightly clippy may raise new warning and I've found it is pretty annoying to have a CI failing for something not yet published.

rursprung commented:
> It seems clippy is not installed on nightly in the CI. But I think it would be better to run clippy only on stable (or make it not mandatory). Nightly clippy may raise new warning and I've found it is pretty annoying to have a CI failing for something not yet published.

good point - i've now changed it to only run for `stable` (and added it to the installation step just in case so that it's always present).

mgautierfr commented:
Looks good to me now !
Thanks for the PR
